src/twisted/conch/ssh/transport.py

```diff
@@ -967,7 +967,7 @@ def ssh_DEBUG(self, packet):
         @type packet: L{bytes}
         @param packet: The message data.
         """
-        alwaysDisplay = bool(packet[0])
```

Comment:
On Python 3 this was correct because indexing a `bytes` object results in the integer value for that index:

```
>>> b'\x00'[0], b'\x01'[0]
(0, 1)
```

On Python 2, however, `bytes` is `str`, so indexing a `bytes` object returns another `bytes` object:

```
>>> b'\x00'[0], b'\x01'[0]
('\x00', '\x01')
```

Because items in the returned tuple are truthy this always evaluated to `True` on Python 2.  Whoops!